pcap-linux.c

```diff
@@ -3298,6 +3300,26 @@ static void map_arphrd_to_dlt(pcap_t *handle, int sock_fd, int arptype,
 		 * others?
 		 */
 		if (!is_wifi(sock_fd, device)) {
+			int ret;
+
+			/* This is not a Wi-Fi device but it could be
+			 * a DSA master/management network device.
+			 */
+			ret = iface_dsa_get_proto_info(device, handle,
+						       handle->errbuf);
+			if (ret < 0)
+				return;
+
+			/* This is a DSA master/management network device
```

Comment:
That moment should go inside the `if (ret == 1) {` block, as the statement in the comment is true only if `iface_dsa_get_proto_info()` returned 1.